stores/metadata.go

```diff
@@ -1125,6 +1125,50 @@ func (s *SQLStore) AddPartialSlab(ctx context.Context, data []byte, minShards, t
 	return s.slabBufferMgr.AddPartialSlab(ctx, data, minShards, totalShards, contractSetID)
 }
 
+func (s *SQLStore) CopyObject(ctx context.Context, srcBucket, dstBucket, srcPath, dstPath string) error {
+	return s.retryTransaction(func(tx *gorm.DB) error {
+		var srcObj dbObject
+		err := tx.Where("objects.object_id = ? AND DBBucket.name = ?", srcPath, srcBucket).
+			Joins("DBBucket").
+			Take(&srcObj).
+			Error
+		if err != nil {
+			return fmt.Errorf("failed to fetch src object: %w", err)
+		}
+
+		var srcSlices []dbSlice
+		err = tx.Where("db_object_id = ?", srcObj.ID).
+			Find(&srcSlices).
+			Error
+		if err != nil {
+			return fmt.Errorf("failed to fetch src slices: %w", err)
+		}
+		for i := range srcSlices {
+			srcSlices[i].Model = Model{} // clear model
+			srcSlices[i].DBObjectID = 0  // clear object id
+		}
+
+		var bucket dbBucket
+		err = tx.Where("name = ?", dstBucket).
+			Take(&bucket).
+			Error
+		if err != nil {
+			return fmt.Errorf("failed to fetch dst bucket: %w", err)
+		}
+
+		dstObj := srcObj
```

Comment:
wondering whether this should be some kind of generic clone using reflection so we don't forget to update this when we extend the object